Library/Homebrew/readall.rb

```diff
@@ -0,0 +1,79 @@
+require "formula"
+require "tap"
+require "thread"
+require "readall"
+
+module Readall
+  class << self
+    def valid_ruby_syntax?(ruby_files)
```

Comment:
Would you mind adding a check that `ruby_files` is a `Queue` instance or alternatively convert it to a queue if it isn't already one? I think it's not obvious on the call site that a `Queue` needs to be supplied and if an `Array` is used, then the code will try to use it (which is not thread-safe), will fortunately fail, but with a rather cryptic error message:

```
TypeError: no implicit conversion of true into Integer
```